EKF/control.cpp

```diff
@@ -1003,26 +1003,37 @@ void Ekf::controlHeightFusion()
 
 	case VDIST_SENSOR_GPS:
 
-		// Determine if GPS should be used as the height source
-		if (do_range_aid && _range_sensor.isDataHealthy()) {
+		// Do switching between GPS and rangefinder if using range finder as
+		// a height source when close to ground and moving slowly
+		if (!_control_status_prev.flags.rng_hgt && do_range_aid && _range_sensor.isDataHealthy()) {
 			setControlRangeHeight();
-			fuse_height = true;
 
 			// we have just switched to using range finder, calculate height sensor offset such that current
 			// measurement matches our current height estimate
 			if (_control_status_prev.flags.rng_hgt != _control_status.flags.rng_hgt) {
 				_hgt_sensor_offset = _terrain_vpos;
 			}
 
-		} else if (!do_range_aid && _gps_data_ready && !_gps_hgt_intermittent && _gps_checks_passed) {
-			fuse_height = true;
+		} else if (_control_status_prev.flags.rng_hgt && !do_range_aid) {
```

Comment:
If we're not using RNG aid, we'll never enter that if and can't fallback to baro. How does that solve the issue?

Reply:
This function is there to select a height sensor and handle switching associated with normal operation.  Fallbacks due to a sensor fusion or data timeout are handled elsewhere in Ekf::controlHeightSensorTimeouts. I will add comment to make this clear.